mitodl/ocw-studio
# Added site creation page

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #45 

#### What's this PR do?
- Adds the site creation page
- Adds the formik library

#### How should this be manually tested?
- Make sure you have at least 1 starter in your db
- Navigate to `/sites/`
- Enter a title and submit

A new website should be created, and you should be redirected to the detail page for that site right away.

#### Any background context you want to provide?
- I removed pagination on the WebsiteStarter API results. We can add it back if we need it, but it seems unlikely that we'd have so many starters that we would really need to page through them
- I changed up the styling of the sidebar + content in the site page. There were some quirks in the way the elements were lining up with the header, and while fixing that I realized that it doesn't make much sense for the sidebar to grow proportionally with the browser window, so I changed it to use flexbox instead of a bootstrap container+columns
- I'm hoping we can establish some patterns for (a) testing forms, and (b) displaying API response errors as form errors. The patterns used in this PR should get some extra scrutiny so we can be sure we're comfortable with them for future use cases.

#### Screenshots (if appropriate)
![ss 2021-02-24 at 15 27 59 ](https://user-images.githubusercontent.com/14932219/109064488-77bf0700-76b8-11eb-97ce-271ad13189b8.png)
![ss 2021-02-24 at 15 49 57 ](https://user-images.githubusercontent.com/14932219/109064493-7ab9f780-76b8-11eb-859b-2c7c278494f4.png)
